soc/arm/atmel_sam0/common/soc_samd2x.c

```diff
@@ -153,11 +153,14 @@ static void gclks_init(void)
 	wait_gclk_synchronization();
 
 	/* OSCULP32K/32 -> GCLK2 */
-	GCLK->GENDIV.reg = GCLK_GENDIV_ID(2) | GCLK_GENDIV_DIV(32 - 1);
+    /* Divider = 2 ^ (GENDIV.DIV + 1) = 32, see 15.8.4 */
+	GCLK->GENDIV.reg = GCLK_GENDIV_ID(2) |
+			   GCLK_GENDIV_DIV(SOC_ATMEL_SAM0_GCLK2_PRESCALER);
 	wait_gclk_synchronization();
 
 	GCLK->GENCTRL.reg =
-	    GCLK_GENCTRL_ID(2) | GCLK_GENCTRL_SRC_OSC32K | GCLK_GENCTRL_GENEN;
+	    GCLK_GENCTRL_ID(2) | GCLK_GENCTRL_SRC_OSCULP32K |
+	    GCLK_GENCTRL_GENEN | GCLK_GENCTRL_DIVSEL;
```

Comment:
Any particular reason to use ultra low power instead XOSC32K?

Reply:
OSCUPL always runs and cannot be turned off.  It has the lowest power consumption as well.  The wdt would normally use this oscillator to prevent situations where the wdt could be accidentally disabled if its oscillator source was disabled.

Reply:
Ok, seems nice!
Do you know if there are differences between WDT_IDs for SAMD20 and SAMR21?

Reply:
@nandojve I am not sure I understand what you mean by the WDT_ID's?  I believe the GCLK GENDIV ID's are the same.  That is, GCLK2 is GCLK2 across all the devices.  Is that what you meant?

Reply:
Hi @sslupsky, forget about my previous comment. Could you check the GENCTRL reg for SAMR21 to confirm if the OSCULP32K is number 3. I would like suggest to people start use defines from headers instead numbers like GCLK_GENCTRL_SRC_OSCULP32K. This is common between all SoC variations and will avoid code issues.